developer/src/kmc-convert/src/keylayout-to-kmn/keylayout-to-kmn-converter.ts

```diff
@@ -1,27 +1,990 @@
+              // from these create an array of modifier combinations  e.g. [['','caps?'], ['Caps']] .....................................................................................................
+              /* eg: [['24', 0], ['24', 3]] */        const b4_deadkey_arr: number[][] = this.get_KeyModifier_array__From__ActionID(jsonObj, action_id);
+              /* e.g. [['','caps?'], ['Caps']]*/      const b4_deadkeyModifier_arr: string[] = this.get_Modifier_array__From__KeyModifier_array(data_ukelele.arrayOf_Modifiers, b4_deadkey_arr);
+              // ........................................................................................................................................................................................
+
+
+              // Data of Block Nr 6 .....................................................................................................................................................................
+              // create an array[action id,state,output] from all state-output-pairs that use state = b5_value_next (e.g. use 1 in  <when state="1" output="â"/> ) ......................................
+              /*  eg: [ 'a9','1','â'] */              const b6_actionId_arr: string[][] = this.get_ActionStateOutput_array__From__ActionState(jsonObj, b5_value_next);
+              // ........................................................................................................................................................................................
+
+
+              // Data of Block Nr 1  ....................................................................................................................................................................
+              // create array[Keycode,Keyname,action id,actionIndex,output] and array[Keyname,action id,behaviour,modifier,output] ......................................................................
+              /*  eg: ['0','K_A','a9','0','â'] */     const b1_keycode_arr: string[][] = this.get_KeyActionOutput_array__From__ActionStateOutput_array(jsonObj, b6_actionId_arr);
+              /*  eg: ['K_A','a9','0','NCAPS','â']*/  const b1_modifierKey_arr: string[][] = this.get_KeyMBehaviourModOutputArray__from__KeyActionBehaviourOutput_array(jsonObj, b1_keycode_arr, isCapsused);
+                // .......................................................................................................................................................................................
+
+                for (let n1 = 0; n1 < b4_deadkeyModifier_arr.length; n1++) {
+                  for (let n2 = 0; n2 < b4_deadkeyModifier_arr[n1].length; n2++) {
+                    for (let n3 = 0; n3 < b4_deadkey_arr.length; n3++) {
+                      for (let n4 = 0; n4 < b1_modifierKey_arr.length; n4++) {
```

Comment:
Whoa, we've got some pretty deep loop nesting here.  Normally, that's not a good sign, but I'm guessing it's necessary due to the format.